extensions/librdkafka/KafkaConnection.h

```diff
@@ -0,0 +1,130 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef NIFI_MINIFI_CPP_KAFKACONNECTION_H
+#define NIFI_MINIFI_CPP_KAFKACONNECTION_H
+
+#include <atomic>
+#include <mutex>
+#include <string>
+#include "core/logging/LoggerConfiguration.h"
+#include "core/logging/Logger.h"
+#include "rdkafka.h"
+#include "KafkaTopic.h"
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace processors {
+
+class KafkaConnectionKey {
+  public:
+    std::string client_id_;
+    std::string brokers_;
+
+    bool operator <(const KafkaConnectionKey& rhs) const {
+      return std::tie(brokers_, client_id_) < std::tie(rhs.brokers_, rhs.client_id_);
+    }
+};
+
+class KafkaConnection {
+ public:
+
+  explicit KafkaConnection(const KafkaConnectionKey &key);
+
+  ~KafkaConnection() {
+    remove();
+  }
+
+  void remove();
+
+  void removeConnection();
+
+  bool initialized() const;
+
+  void setConnection(rd_kafka_t *producer, rd_kafka_conf_t *conf);
+
+  rd_kafka_conf_t *getConf();
+
+  rd_kafka_t *getConnection();
+
+  bool hasTopic(const std::string &topic) const;
+
+  std::shared_ptr<KafkaTopic> getTopic(const std::string &topic);
+
+  KafkaConnectionKey const * const getKey() const;
+
+  void putTopic(const std::string &topicName, const std::shared_ptr<KafkaTopic> &topic);
+
+  static void logCallback(const rd_kafka_t* rk, int level, const char* /*fac*/, const char* buf);
+
+  bool tryUse();
+
+  friend class KafkaLease;
+
+ private:
+
+  std::shared_ptr<logging::Logger> logger_;
+
+  std::mutex mutex_;
+
+  bool lease_;
+
+  std::atomic<bool> initialized_;
```

Comment:
This seems to be a leftover. As the class is not thread-safe, having such a member is misleading, please keep it as a simple bool. Atomic include can go away as well.